netmiko/base_connection.py

```diff
@@ -386,10 +388,18 @@ def __init__(
                     no_log=no_log,
                     record_writes=session_log_record_writes,
                 )
+            elif isinstance(session_log, SessionLog):
+                # SessionLog object
+                self.session_log = session_log
+                # Add filter after self.session_log is constructed
+                log.addFilter(SecretsFilter(no_log=self.session_log.no_log))
```

Comment:
I think we should decouple this, not have Netmiko does this automatically (i.e. the user should do this in their code if they want to set the Python logging the same way as the session_log).

End-user should be able to do this:

```python
>>> import netmiko
>>> netmiko.log
<Logger netmiko (WARNING)>
>>> from netmiko.base_connection import SecretsFilter
>>> netmiko.log.addFilter(SecretsFilter(no_log=log_dict))
```

Reply:
Agreed. I pushed a change to remove the filter.